services/keyword/create_keyword_result_test.go

```diff
@@ -0,0 +1,390 @@
+				err := ormer.QueryTable(Keyword{}).RelatedSel().One(keywordRecord)
+				if err != nil {
+					Fail(fmt.Sprintf("Get first record failed: %v", err.Error()))
+				}
+
+				Expect(keywordRecord).NotTo(BeNil())
+				Expect(keywordRecord.Id).To(BeNumerically(">", 0))
+				Expect(keywordRecord.Keyword).To(Equal(keyword))
+
+				Expect(keywordRecord.User.Id).To(Equal(user.Id))
+			})
+
+			It("saves raw HTML to page table", func() {
+				position := FabricatePosition(faker.Word(), faker.Word(), faker.Word())
+				user := FabricateUser(faker.Email(), faker.Password())
+				keyword := faker.Word()
+				url := fmt.Sprintf("https://www.google.com/search?q=%s&lr=lang_en", keyword)
+				rawHtml := faker.Paragraph()
+				linkList := []Link{
+					{
+						Url:      fmt.Sprintf("https://www.google.com/search?q=%s", faker.Word()),
+						Position: position,
+					},
+				}
+
+				createKeywordResultService := service.CreateKeywordResult{
+					Keyword:  keyword,
+					Url:      url,
+					LinkList: linkList,
+					RawHtml:  rawHtml,
+					User:     user,
+				}
+
+				keywordRecord, _ := createKeywordResultService.Run()
+
+				ormer := orm.NewOrm()
+				pageRecord := &Page{}
+				err := ormer.QueryTable(Page{}).RelatedSel().One(pageRecord)
+				if err != nil {
+					Fail(fmt.Sprintf("Get first record failed: %v", err.Error()))
+				}
+
+				Expect(pageRecord).NotTo(BeNil())
+				Expect(pageRecord.Id).To(BeNumerically(">", 0))
+				Expect(pageRecord.RawHtml).To(Equal(rawHtml))
+
+				Expect(pageRecord.Keyword.Id).To(Equal(keywordRecord.Id))
+			})
+
+			It("saves all links to link table", func() {
+				position := FabricatePosition(faker.Word(), faker.Word(), faker.Word())
+				user := FabricateUser(faker.Email(), faker.Password())
+				keyword := faker.Word()
+				url := fmt.Sprintf("https://www.google.com/search?q=%s&lr=lang_en", keyword)
+				rawHtml := faker.Paragraph()
+				linkList := []Link{
+					{
+						Url:      fmt.Sprintf("https://www.google.com/search?q=%s", faker.Word()),
+						Position: position,
+					},
+				}
+
+				createKeywordResultService := service.CreateKeywordResult{
+					Keyword:  keyword,
+					Url:      url,
+					LinkList: linkList,
+					RawHtml:  rawHtml,
+					User:     user,
+				}
+
+				keywordRecord, _ := createKeywordResultService.Run()
+
+				ormer := orm.NewOrm()
+				linkRecord := &Link{}
+				err := ormer.QueryTable(Link{}).RelatedSel().One(linkRecord)
+				if err != nil {
+					Fail(fmt.Sprintf("Get first record failed: %v", err.Error()))
+				}
+
+				Expect(linkRecord).NotTo(BeNil())
+				Expect(linkRecord.Id).To(BeNumerically(">", 0))
+				Expect(linkRecord.Url).To(Equal(linkList[0].Url))
+
+				Expect(linkRecord.Keyword.Id).To(Equal(keywordRecord.Id))
+				Expect(linkRecord.Position.Id).To(Equal(position.Id))
+			})
+		})
+
+		Context("given INVALID params", func() {
+			It("does NOT return a keyword record", func() {
+				overLengthKeyword := faker.Word() + strings.Repeat("*", 500)
```

Comment:
There could be two contexts as there cane b two possible invalid params:

- Too long keyword
- Too long URL

```suggestion
Context("given INVALID params", func() {
            Context("given an INVALID keyword", func() {})
           	Context("given an INVALID URL", func() {})

```

Or both cases could be done in one test.